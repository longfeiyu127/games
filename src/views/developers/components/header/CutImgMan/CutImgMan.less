@count: 20;

@radius: 20;

@random: `Math.random() `;

@0px: 0px;
@0deg: 0deg;

@sunspotsColor: rgba(137, 107, 37, 0.6);

.getShadow(@a) {
  @top: floor(sin(@a) * @radius * @random) + @0px;
  @left: floor(cos(@a) * @radius * @random) + @0px;
  box-shadow+: @left @top @sunspotsColor;
}

.getSunspots(@i) when (@i < @count) {
  @a: @random*3.1415*2;
  .getShadow(@a);
  .getSunspots(@i+1);
}

.c-cutImgMan {
  position: absolute;
  left: 50%;
  top: 150px;
  transform: translate(-50%);
  .start-box {
    position: absolute;
    height: 50px;
    width: 50px;
    top: 70px;
    z-index: 1;
    left: 50%;
    animation: inflation linear 2s infinite;

    @keyframes inflation {
      0%,
      100% {
        transform: translate(-50%) scale(1);
      }
      50% {
        transform: translate(-50%) scale(1.1);
      }
    }
    .star {
      position: absolute;
      height: 50px;
      width: 50px;
      border-radius: 50%;
      background: #f9fa71;
      box-shadow: inset 0px 0px 3px 2px rgba(255, 255, 255, 1), inset 0px 0px 15px rgba(255, 255, 255, 1),
        0px 0px 5px 2px #ffffff, 0px 0px 40px 5px #fdfc55, 0px 0px 10px 3px #fdfc55;
      .sunspots {
        position: absolute;
        height: 2px;
        width: 2px;
        border-radius: 50%;
        left: 25px;
        top: 25px;
        background: @sunspotsColor;
        animation: flow linear 20s infinite alternate;
        filter: blur(2px) contrast(2);
        .getSunspots(0);
      }
    }
    @minlength: 80;
    @maxlength: 60;
    .getrRadiation() {
      width: 140px;
      height: 2px;
      border-radius: 50%;
      position: absolute;
      left: 50%;
      top: 50%;
      transform: translate(-50%, -50%);
      background: #c8c871;
      z-index: 0;
      // box-shadow: 0px 0px 10px 3px #fdfc55;
      background: linear-gradient(to right, rgba(200, 200, 113, 0), rgba(200, 200, 113, 1), rgba(200, 200, 113, 0));
    }
    .radiation {
      .getrRadiation();
    }
    .getRam() {
      width: @minlength + floor(@random* @maxlength) + @0px;
      @Angle: floor(@random*360) + @0deg;
      transform-origin: 50% 50%;
      transform: translate(-50%, -50%) rotate(@Angle);
    }
    .radiation1 {
      .getRam();
    }
    .radiation2 {
      .getRam();
    }
    .radiation3 {
      .getRam();
    }
    .radiation4 {
      .getRam();
    }
    .radiation5 {
      .getRam();
    }
    .radiation6 {
      .getRam();
    }
    .radiation7 {
      .getRam();
    }
    .radiation8 {
      .getRam();
    }
  }
}

@keyframes flow {
  to {
    .getSunspots(0);
  }
}
