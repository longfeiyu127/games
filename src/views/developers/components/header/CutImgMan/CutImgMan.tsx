import React from 'react'
import './CutImgMan.less'
import cutImgMan from '@/assets/images/developers/cutImg-man.jpg'
import HeaderImg from '../HeaderImg/HeaderImg'

const CutImgMan = () => (
  <div className="c-cutImgMan">
    <HeaderImg img={cutImgMan} />
    <div className="start-box">
      <div className="radiation radiation1" />
      <div className="radiation radiation2" />
      <div className="radiation radiation3" />
      <div className="radiation radiation4" />
      <div className="radiation radiation5" />
      <div className="radiation radiation6" />
      <div className="radiation radiation7" />
      <div className="radiation radiation8" />
      <div className="star">
        <div className="sunspots sunspots" />
      </div>
    </div>
  </div>
)

export default CutImgMan
