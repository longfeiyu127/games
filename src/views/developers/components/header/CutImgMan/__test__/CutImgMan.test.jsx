import React from 'react'
import renderer from 'react-test-renderer'
import CutImgMan from '../CutImgMan.tsx'

describe('<CutImgMan />', () => {
  it('component developers/header/CutImgMan render', () => {
    const Dom = renderer.create(<CutImgMan />).toJSON()
    expect(Dom).toMatchSnapshot()
  })
})
