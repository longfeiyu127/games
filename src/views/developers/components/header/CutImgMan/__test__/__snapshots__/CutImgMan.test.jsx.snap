// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CutImgMan /> component developers/header/CutImgMan render 1`] = `
<div
  className="c-cutImgMan"
>
  <div
    className="c-header-img"
    style={Object {}}
  >
    <img
      alt="header"
      src="cutImg-man.jpg"
    />
  </div>
  <div
    className="start-box"
  >
    <div
      className="radiation radiation1"
    />
    <div
      className="radiation radiation2"
    />
    <div
      className="radiation radiation3"
    />
    <div
      className="radiation radiation4"
    />
    <div
      className="radiation radiation5"
    />
    <div
      className="radiation radiation6"
    />
    <div
      className="radiation radiation7"
    />
    <div
      className="radiation radiation8"
    />
    <div
      className="star"
    >
      <div
        className="sunspots sunspots"
      />
    </div>
  </div>
</div>
`;
